### PR TITLE
Journalist app refactor (the remix)

### PR DESCRIPTION
**Pull request by heartsucker**
## Status

Ready for merge

## Description of Changes

Refactor the journalist interface like how we did to the source interface in #2365.

This only does a few things because so many of the tests break because of excessive mocking and working outside the application context. I think originally we mocked to get around the app context errors when in reality we just need a `with app.app_context()` in a few places.

## Testing

`pytest -vx tests/` and do some manual checking if you don't trust the test suite. Since this affects the AppArmor rule, staging should be booted and used for testing.

## Deployment

Need to ensure AppArmor rules and rsync filter are accurate.

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

### If you made changes to the system configuration:

- [x] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### CODE REVIEW

File: install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2 [hunk 1]
```diff
@@ -129,6 +129,11 @@
   /var/www/securedrop/dictionaries/nouns.txt r,
   /var/www/securedrop/journalist.py r,
   /var/www/securedrop/journalist.pyc rw,
+  /var/www/securedrop/journalist_app/__init__.py r,
+  /var/www/securedrop/journalist_app/__init__.pyc rw,
+  /var/www/securedrop/journalist_app/utils.py r,
+  /var/www/securedrop/journalist_app/utils.pyc rw,
```
Comment (dachary) [inline install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2]: missing forms.py here ?

**Comment by heartsucker** [comment]
@dachary Can you re-review or dismiss your stale review?

**Comment by dachary** [comment]
I don't know how to discard my review ? So I approved it instead to not block things. But I'm yet to review the PR.

**Comment by dachary** [comment]
unless someone objects, I'll review this fine commit series in the next 24h

**Comment by heartsucker** [comment]
@dachary If this ever hits a point where everything passes, it should be good to merge, so we could get it into `develop`. However, so long as there is anything left in `journalist.py`, I'm going to keep moving files over to `journalist_app/`. You might want to wait.

**Comment by dachary** [comment]
@heartsucker I'm happy to wait ;-)

File: install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2 [hunk 2]
```diff
@@ -129,6 +129,23 @@
   /var/www/securedrop/dictionaries/nouns.txt r,
   /var/www/securedrop/journalist.py r,
   /var/www/securedrop/journalist.pyc rw,
+  /var/www/securedrop/journalist_app/__init__.py r,
+  /var/www/securedrop/journalist_app/__init__.pyc rw,
```
Comment (msheiny) [inline install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2]: this is a nit (kind of talking outloud here - you certainly dont need to trial it for this PR) -- but I wonder if we can do PCRE notation on these ... 
looks like:  `var/www/securedrop/journalist_app/__init__.{pyc,py} rw` might work....

http://wiki.apparmor.net/index.php/QuickProfileLanguage

Comment (msheiny) [reply install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2]: oh im dumb -- missed the alternating `r` / `rw`s

Comment (heartsucker) [reply install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2]: I think it's a good thing that the app can't write to it's own source files, actually.

Comment (heartsucker) [reply install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2]: Principle of least privilege 'n all.

Comment (heartsucker) [reply install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/files/usr.sbin.apache2]: 🐩

File: securedrop/tests/pages-layout/test_journalist.py [hunk 3]
```diff
@@ -104,13 +104,13 @@ def test_admin_new_user_two_factor_hotp(self):
...
         self._admin_creates_a_user(hotp='123456')
-        self._screenshot('journalist-admin_new_user_two_factor_hotp.png')
+        self._screenshot('journalist-admin.new_user_two_factor_hotp.png')
 
     def test_admin_new_user_two_factor_totp(self):
         self._admin_logs_in()
         self._admin_visits_admin_interface()
         self._admin_creates_a_user(hotp=None)
-        self._screenshot('journalist-admin_new_user_two_factor_totp.png')
+        self._screenshot('journalist-admin.new_user_two_factor_totp.png')
```
Comment (redshiftzero) [inline securedrop/tests/pages-layout/test_journalist.py]: Let's not rename any screenshots - this will break the screenshot autoupdating functionality

**Pull request merged**